R/git.R

```diff
@@ -101,12 +101,41 @@ git_branches_no_merged <- function(root = NULL, include_master = FALSE) {
                       root = root, check = TRUE)$output
     branches <- c(master, branches)
   }
-  branches <- read.table(text = branches, stringsAsFactors = FALSE, sep = ",",
-                         col.names = c("name", "last_commit"))
+  branches <- utils::read.table(text = branches, stringsAsFactors = FALSE,
+                                sep = ",", col.names = c("name", "last_commit"))
   branches <- branches[branches$name != "gh-pages", ]
-  branches$last_commit_age <-
-    rep(as.integer(Sys.time()), nrow(branches)) - branches$last_commit
-  times <- as.POSIXct(branches$last_commit, origin = "1970-01-01", tz = "UTC")
-  branches$last_commit <- strftime(times)
+  branches$last_commit_age <- calculate_age(branches$last_commit)
+  branches$last_commit <- convert_unix_to_iso_time(branches$last_commit)
   branches
 }
+
+## This gets last 25 commits from master
+## if not master then gets the unmerged commits (limit 25)
+git_commits <- function(branch, root = NULL) {
+  if (branch == "master") {
+    args <- c("log", "--pretty='%h,%cd'", "--date=unix", "--max-count=25",
+              sprintf("refs/remotes/origin/%s", branch))
+  } else {
+    remote_branch <- sprintf("refs/remotes/origin/%s", branch)
+    args <- c("log", "--pretty='%h,%cd'", "--date=unix", "--max-count=25",
```

Comment:
nice - just checked and this seems to get exactly the right sort of thing on unmerged branches that have diverged from master